packages/jest-runner/src/index.ts

```diff
@@ -15,28 +15,24 @@ import type {
   TestResult,
 } from '@jest/test-result';
 import {deepCyclicCopy} from 'jest-util';
+import type {TestWatcher} from 'jest-watcher';
 import {PromiseWithCustomMessage, Worker} from 'jest-worker';
 import runTest from './runTest';
 import type {SerializableResolver, worker} from './testWorker';
-import {
-  EmittingTestRunner,
-  TestRunnerOptions,
-  TestWatcher,
-  UnsubscribeFn,
-} from './types';
+import {EmittingTestRunner, TestRunnerOptions, UnsubscribeFn} from './types';
 
 const TEST_WORKER_PATH = require.resolve('./testWorker');
 
 interface WorkerInterface extends Worker {
   worker: typeof worker;
 }
 
+export type {TestWatcher} from 'jest-watcher';
```

Comment:
Reexport simplifies typing a custom runner. See example in type test, there is no need to import `jest-watcher`. Or am I overthinking?